integration_test.go

```diff
@@ -283,3 +286,89 @@ func TestDefaultValue(t *testing.T) {
 		return result.ID > 0
 	}, "ID should be greater than 0")
 }
+
+func TestDistinct(t *testing.T) {
+	skipIfShort(t)
+	t.Parallel()
+	dsn, cleanup, err := createTestDB(context.Background())
+	if err != nil {
+		log.Fatalf("could not init integration tests while creating database: %v", err)
+		os.Exit(1)
+	}
+	defer cleanup()
+	// Open db.
+	db, err := gorm.Open(New(Config{
+		DriverName: "spanner",
+		DSN:        dsn,
+	}), &gorm.Config{PrepareStmt: true})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := db.AutoMigrate(&testutil.User{}, &testutil.Account{}, &testutil.Pet{}, &testutil.Company{}, &testutil.Toy{}, &testutil.Language{},
+		&testutil.Coupon{}, &testutil.CouponProduct{}, &testutil.Order{}, &testutil.Parent{}, &testutil.Child{}); err != nil {
+		t.Fatalf("Failed to migrate models, got error: %v", err)
+	}
+
+	users := []testutil.User{
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct-2", testutil.Config{}),
+		*testutil.GetUser("distinct-3", testutil.Config{}),
+	}
+	users[0].Age = 20
+
+	if err := db.Create(&users).Error; err != nil {
+		t.Fatalf("errors happened when create users: %v", err)
+	}
+
+	var names []string
+	db.Table("users").Where("name like ?", "distinct%").Order("name").Pluck("name", &names)
+	testutil.AssertEqual(t, names, []string{"distinct", "distinct", "distinct", "distinct-2", "distinct-3"})
+
+	var names1 []string
+	db.Model(&testutil.User{}).Where("name like ?", "distinct%").Distinct().Order("name").Pluck("Name", &names1)
+
+	testutil.AssertEqual(t, names1, []string{"distinct", "distinct-2", "distinct-3"})
+
+	var names2 []string
+	db.Scopes(func(db *gorm.DB) *gorm.DB {
+		return db.Table("users")
+	}).Where("name like ?", "distinct%").Order("name").Pluck("name", &names2)
+	testutil.AssertEqual(t, names2, []string{"distinct", "distinct", "distinct", "distinct-2", "distinct-3"})
+
+	var results []testutil.User
+	if err := db.Distinct("name", "age").Where("name like ?", "distinct%").Order("name, age desc").Find(&results).Error; err != nil {
+		t.Errorf("failed to query users, got error: %v", err)
+	}
+
+	expects := []testutil.User{
+		{Name: "distinct", Age: 20},
+		{Name: "distinct", Age: 18},
+		{Name: "distinct-2", Age: 18},
+		{Name: "distinct-3", Age: 18},
+	}
+
+	if len(results) != 4 {
+		t.Fatalf("invalid results length found, expects: %v, got %v", len(expects), len(results))
+	}
+
+	for idx, expect := range expects {
+		testutil.AssertObjEqual(t, results[idx], expect, "Name", "Age")
+	}
+
+	var count int64
+	if err := db.Model(&testutil.User{}).Where("name like ?", "distinct%").Count(&count).Error; err != nil || count != 5 {
```

Comment:
Instead of saying it as `5` directly, 
can we do something like `expectedNames := []string{"distinct", "distinct", "distinct", "distinct-2", "distinct-3"}`
and then use
```
count != len(expectedNames)
```.
This `expectedNames` can also be used above at some places instead of hardcoding. This will make it clear from where the 5 is coming.